src/fetchServices.ts

```diff
@@ -0,0 +1,171 @@
+}
+
+export async function listServices(
+  client: servicemanagement_v1.Servicemanagement,
+  serviceNames: string[],
+  nextPageToken = ''
+): Promise<string[]> {
+  const res = await client.services.list(
+    {
+      pageSize: 100,
+      pageToken: nextPageToken,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+
+  if (res.data.services) {
+    res.data.services.forEach(item => {
+      if (item.serviceName) {
+        serviceNames.push(item.serviceName);
+      }
+    });
+  }
+
+  if (res.data.nextPageToken) {
+    await listServices(client, serviceNames, res.data.nextPageToken);
+  }
+  return serviceNames;
+}
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.getConfig endpoint
+ * @param serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return the service config of the passed hostname
+ */
+export async function getServiceConfig(
+  serviceName: string
+): Promise<servicemanagement_v1.Schema$Service> {
+  const res = await servicemanagement.services.getConfig(
+    {
+      serviceName: serviceName,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+  return res.data;
+}
+
+/**
+ * Determines if a given service is a 'Cloud API'.
+ * @param serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return well is it, or not?
+ */
+export async function isCloudApi(
+  serviceConfig: servicemanagement_v1.Schema$Service
+): Promise<boolean> {
+  if (serviceConfig.title?.includes('Firebase')) {
+    return false;
+  }
+  if (serviceConfig.authentication?.rules) {
+    const scopes = serviceConfig.authentication.rules.map(item => {
+      if (item.oauth?.canonicalScopes) {
+        const scopesStr: string = item.oauth.canonicalScopes;
+        if (scopesStr.includes('auth/firebase')) {
+          return false;
+        } else if (scopesStr.includes('auth/cloud-platform')) {
+          return true;
+        } else if (scopesStr.includes('auth/drive')) {
+          return true;
+        } else if (scopesStr.includes('auth/apps')) {
+          return true;
+        }
+      }
+      return false;
+    });
+
+    if (scopes.includes(true)) {
+      return true;
+    }
+  }
+
+  // usage: { requirements: [ 'serviceusage.googleapis.com/tos/cloud' ] }
+  if (serviceConfig.usage?.requirements) {
+    const usage: string[] = serviceConfig.usage.requirements;
+    if (usage.includes('serviceusage.googleapis.com/tos/cloud')) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Classify all available services as Cloud API or not and export to a known Google Sheet.
+ */
+export async function exportApisToSheets() {
+  const services: string[] = await getAllServiceNames();
+  const values: string[][] = await Promise.all(
+    services.map(async s => {
+      const config = await getServiceConfig(s);
+      return [s, String(config.title), String(await isCloudApi(config))];
```

Comment:
I actually have no idea what `String` will do 😆   Will just using `config.title` work?  No casting necessary.

Reply:
It won't work naked - it's typed from here: https://github.com/googleapis/google-api-nodejs-client/blob/master/src/apis/servicemanagement/v1.ts#L1490 so it barfs at me as being (string | null | undefined)

Reply:
No biggie, but usually you would say `config.title || ''` if you needed to get an empty string in that case instead of undefined or null :)